OSDOCS-11760: Admin/Dev console 4.17 release notes (#82377)
openshift/openshift-docs: release_notes/ocp-4-17-release-notes.adoc

[id="ocp-4-17-web-console_{context}"]
=== Web console

[id="ocp-4-17-ocp-lightspeed_{context}"]
==== {ols} is available in the web console

> Starting with {product-title} 4.16, {ols} is available for use in the web console.

Reviewer: Shall we add some new features about OLS here or in `Administrator perspective` section? 
For example [PR](https://github.com/openshift/console/pull/14030) addes Lightspeed hover button and user preference settings to show/hide lightspeed button

> Starting with {product-title} 4.16, {ols} is available for use in the web console.

Reviewer: ```suggestion
Starting with {product-title} 4.16, {ols} operator is available for use in the web console.
```

> Starting with {product-title} 4.16, {ols} Operator is available for use in the web console. With this release, a hover button was added to help you discover {ols}. Once you click the hover button, the chat window will appear with instructions on how to enable and install {ols} on the cluster. You can hide the {ols} button when changing the default user preferences.

Reviewer 2: What is a hover button?

Reviewer 2: ```suggestion
Starting with {product-title} 4.16, {ols} Operator is available for use in the web console. With this release, a hover button was added to help you discover {ols}. Once you click the hover button, the chat window appears with instructions on how to enable and install {ols} on the cluster. You can hide the {ols} button when changing the default user preferences.
```

//https://github.com/openshift/openshift-docs/pull/81581
//For more information, see xref:../web_console/capabilities_products-web-console.html#openshift-lightspeed-web-console_capabilities-web-console[Enhancing your experience in the {product-title} web console by using {ols-official}].

[id="ocp-4-17-administrator-perspective_{context}"]
==== Administrator perspective

This release introduces the following updates to the *Administrator* perspective of the web console:

> * Deprecated Operators are displayed in the OperatorHub before and after installation.

Reviewer: ```suggestion
* Deprecated Operators will be displayed along with a warning icon in the OperatorHub before and after installation.
```

Reviewer: We may need to clarify what's the difference between deprecated operators and normal operators(made a suggestion above)
![Screenshot 2024-09-25 at 10 37 15 AM](https://github.com/user-attachments/assets/cba74f51-9c2f-4b2c-b408-763253da017a)

> * Config files for `MachineConfig` objects were moved to a new tab enabling you to check the content of the configuation files without having to manually retrieve its contents.

Reviewer: ```suggestion
* Config files for `MachineConfig` objects were moved to a new section enabling you to check the content of the configuation files without having to manually retrieve its contents.
```

Reviewer: ![Screenshot 2024-09-25 at 10 42 24 AM](https://github.com/user-attachments/assets/52a37546-5dc6-42a8-a0db-a05cfde01857)

> * An alert was added to `operator-hub-item-details` and `operator-hub-subscribe` components if your cluster is on {gcp-first} with Workload Identity Foundation (WIF).

Reviewer: Shall we use pages instead of component names in release note?

> * An alert was added to `operator-hub-item-details` and `operator-hub-subscribe` components if your cluster is on {gcp-first} with Workload Identity Foundation (WIF).

Reviewer: ```suggestion
* An alert will be shown on Operator details and Operator installation page if your cluster is on {gcp-first} with Workload Identity Foundation (WIF).
```

Reviewer: **operator details modal**
![Screenshot 2024-09-25 at 10 48 04 AM](https://github.com/user-attachments/assets/1ca46679-b761-4c3b-8551-647361787214)

**operator installation page**
![Screenshot 2024-09-25 at 10 48 09 AM](https://github.com/user-attachments/assets/593b211b-a0e9-4beb-b2b3-85fea83b8a8e)

Author: I see what you are saying! Okay, that makes sense! Thanks!

> * Deprecated Operators are displayed in the OperatorHub before and after installation along with a warning notification the Operator is deprecated.

Reviewer 2: ```suggestion
* Deprecated Operators are displayed in the OperatorHub before and after installation along with a warning notification that the Operator is deprecated.
```

> * A page for ShipWright `BuildStrategy` was added with a `ClusterBuildStrategy` and `BuildStrategy` tab.

Reviewer: Is this Dev Console new change? admin console doesn't have this new feature in 4.17

Author: I see it was added with this PR https://github.com/openshift/console/pull/14029

@sanketpathak could you weigh in on this?

Reviewer 3: Yes this feature is added as part of epic [ODC-7518](https://issues.redhat.com/browse/ODC-7518)

Reviewer 2: Will users know where to look for the new page?

> * Config files for `MachineConfig` objects were moved to a new section enabling you to check the content of the configuation files without having to manually retrieve its contents.

Reviewer 2: Where were they moved to?

[id="ocp-4-17-customize-create-project-modal-dyanmic-plugin"]
===== Customize the *Create project* modal using dynamic plugins

> With this release, a new extension point was added, so dynamic plugin creators can pass a component that renders in place of the default *Create project* modal.

Reviewer: ```suggestion
With this release, a new extension point was added, so dynamic plugin creators can pass a component that renders in place of the default *Create Project* modal.
```
![Screenshot 2024-09-25 at 10 50 11 AM](https://github.com/user-attachments/assets/daa4e9d7-3ea8-4d4f-9b9b-b3faaa5d4603)

For more information on {product-title} Console dynamic plugin SDK extensions, see xref:../web_console/dynamic-plugin/dynamic-plugins-reference.adoc#dynamic-plugin-sdk-extensions_dynamic-plugins-reference[Dynamic plugin extension types].

[id="ocp-4-17-console-functional-external-oidc-issuer"]
===== External OpenID Connect (OIDC) token issuer is now functional in the web console

> With this update, the web console works as expected when internal `oauth-server` and `oauth-apiserver` are removed and replaced with an external OpenID Connect (OIDC) issuer.

Reviewer 2: `oauth-server` and `oauth-apiserver` Each needs a noun.

[id="ocp-4-17-developer-perspective_{context}"]
==== Developer Perspective

This release introduces the following updates to the *Developer* perspective of the web console:

> * When you use one of the import flows to create a new deployment, the *Import from Git* or *Container images* automatically opens them in the sidebar.

Reviewer 3: ```suggestion
* When you use one of the add flows to create a new deployment, the *Import from Git* or *Container images* automatically opens them in the sidebar.
```

> * Easily select the desired *Git Type* without having to use the list if {product-title} is unable to identify it.

Reviewer 2: Make this a complete sentence:
```suggestion
* You can easily select the desired *Git Type* without having to use the list if {product-title} is unable to identify it.
```

Reviewer 2: Also, ambiguous _it_. And will users know what a Git Type is and where to select it?

* *Import from Git* supports GitEA, an open-source alternitve to GitHub.
* A warning notification displays on the *Topology* page if the `PodDisruptionBudget` limit is reached.

> * Use Shipwright Builds when importing applications through the *Import from Git* flows to use Shipwright Build strategies such as buildpacks when building your application.

Reviewer 3: ```suggestion
* Use Shipwright Builds when importing applications through the *Import from Git* flows to use Shipwright Build strategies such as S2I, buildpack and Dockerfile (buildah) strategy for building the image```

> * Use Shipwright Builds when importing applications through the *Import from Git* flows to use Shipwright Build strategies such as buildpacks when building your application.

Reviewer 2: Make this a complete sentence:
```suggestion
* You can use Shipwright Builds when importing applications through the *Import from Git* flows to use Shipwright Build strategies such as buildpacks when building your application.
```

Reviewer 2: Maybe split this into two sentences? 
```suggestion
* You can use Shipwright Builds when importing applications through the *Import from Git* flows. Also, you can use Shipwright Build strategies such as buildpacks when building your application.
```
